quill-core/src/main/scala/io/getquill/context/ActionMacro.scala

```diff
@@ -100,7 +100,7 @@ class ActionMacro(val c: MacroContext)
     q"""
       expanded.ast match {
         case io.getquill.ast.Returning(_, _, io.getquill.ast.Property(_, property)) => 
-          property
+          expanded.naming.column(property)
```

Comment:
I think @mentegy is right.  It doesn't apply `NamingStrategy` because the translation of `Returning` is actually happened outside of `Idiom`